File: adxl345_test/src/main/java/com/plattysoft/adxl345_test/MainActivity.java
package com.plattysoft.adxl345_test;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import com.plattysoft.adxl345.ADXL345;

import java.io.IOException;
import java.util.Timer;
import java.util.TimerTask;

public class MainActivity extends Activity {
    private static final String TAG = MainActivity.class.getSimpleName();
    private static final String bus = "I2C1";

    ADXL345 adxl345 = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            adxl345 = new ADXL345(bus);
        }
        catch (IOException e) {
            e.printStackTrace();
        }
        Timer t = new Timer();
        t.schedule(new TimerTask() {
            @Override
            public void run() {
                getSensorData();
            }
        }, 200, 200);
    }

    private void getSensorData(){
        try {
            Log.i(TAG , " Accel X: "+String.valueOf(adxl345.getAccelerationX())
                    +", Accel Y: "+ String.valueOf(adxl345.getAccelerationY())
                    +", Accel Z: "+ String.valueOf(adxl345.getAccelerationZ()));
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        try {
            if (adxl345 != null) {
                adxl345.close();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

}

File: pcf8575_sample/src/main/java/com/plattysoft/pcf8574_sample/MainActivity.kt
package com.plattysoft.pcf8574_sample

import android.app.Activity
import android.os.Bundle
import android.util.Log
import com.google.android.things.pio.Gpio
import com.google.android.things.pio.GpioCallback
import com.plattysoft.pcf8575.Pcf8575
import java.util.*
import kotlin.concurrent.timerTask

private val TAG = MainActivity::class.java.simpleName

class MainActivity : Activity() {

    private val mGpioBoard = Pcf8575.open("BCM23")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var current = 12
        var value = true
        val button = mGpioBoard.openGpio("P00")
        button.setDirection(Gpio.DIRECTION_IN)
        button.registerGpioCallback(GpioCallback() {
            Log.d("PCF8575","Read value (INT): "+it.getValue()+" - "+System.currentTimeMillis())
            true
        })
        val t = Timer()
        t.schedule(timerTask {
            mGpioBoard.setValue(current, value)
            current++;
            if (current >= 16) {
                current = 12
                value = !value
            }
            Log.d("PCF8575","Read value: "+button.getValue()+" - "+System.currentTimeMillis())

        }
        , 200, 500)
    }

    override fun onDestroy() {
        super.onDestroy()
        mGpioBoard.close()
    }

}
